Week7/app/src/main/java/com/bartoszkrol/catfacts/viewmodel/CatFactsViewModel.kt

```diff
@@ -0,0 +1,35 @@
+package com.bartoszkrol.catfacts.viewmodel
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.viewModelScope
+import com.bartoszkrol.catfacts.database.CatFactsDatabase
+import com.bartoszkrol.catfacts.database.RoomRepository
+import com.bartoszkrol.catfacts.model.CatFact
+import kotlinx.coroutines.launch
+
+class CatFactsViewModel(application: Application) : AndroidViewModel(application) {
+
+    private val repository: RoomRepository
+
+    private val allCatFacts: LiveData<List<CatFact>>
+
+    init {
+        val catFactsDao = CatFactsDatabase.getDatabase(application).catFactsDao()
```

Comment:
Avoid initializing dependencies in classes that use them. If they change, you have to change all the classes too.

Use the App class to provide dependencies instead, and pass them in through the constructor!